src/quacc/recipes/espresso/core.py

```diff
@@ -0,0 +1,133 @@
+from quacc.utils.files import load_yaml_calc
+from quacc.calculators.espresso.io import parse_pp_and_cutoff
+
+ESPRESSO_CMD = f"{SETTINGS.ESPRESSO_CMD}"
+ESPRESSO_PP_PATH = f"{SETTINGS.ESPRESSO_PP_PATH}"
+ESPRESSO_PRESET_PATH = f"{SETTINGS.ESPRESSO_PRESET_PATH}"
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from ase import Atoms
+
+    from quacc.schemas._aliases.ase import RunSchema
+
+
+@job
+def static_job(
+    atoms: Atoms,
+    preset: str | None = None,
+    copy_files: list[str] | None = None,
+    **kwargs,
+) -> RunSchema:
+    """
+    Function to carry out a single-point calculation.
+
+    Parameters
+    ----------
+    atoms
+        Atoms object
+    **kwargs
+        Custom kwargs for the espresso calculator. Set a value to
+        `None` to remove a pre-existing key entirely. For a list of available
+        keys, refer to the `ase.calculators.espresso.Espresso` calculator.
+
+        !!! Info "Calculator defaults"
+
+            ```python
+            {
+                "ecutwfc": 40,
+                "ecutrho": 160,
+                "nspin": 1,
+            }
+            ```
+
+    Returns
+    -------
+    RunSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][]
+    """
+
+    calc_defaults = {
+        "ecutwfc": 40,
+        "ecutrho": 160,
+        "nspin": 1,
+    }
+    return _base_job(
+        atoms,
+        calc_defaults=calc_defaults,
+        calc_swaps=kwargs,
+        additional_fields={"name": "espresso Static"},
+        copy_files=copy_files,
+    )
+
+
+def _base_job(
+    atoms: Atoms,
+    preset: str | None = None,
+    calc_defaults: dict[str, Any] | None = None,
+    calc_swaps: dict[str, Any] | None = None,
+    additional_fields: dict[str, Any] | None = None,
+    copy_files: list[str] | None = None,
+) -> RunSchema:
+    """
+    Base function to carry out espresso recipes.
+
+    Parameters
+    ----------
+    atoms
+        Atoms object
+    calc_defaults
+        The default calculator parameters.
+    calc_swaps
+        Custom kwargs for the espresso calculator. Set a value to
+        `None` to remove a pre-existing key entirely. For a list of available
+        keys, refer to the `ase.calculators.espresso.Espresso` calculator.
+    additional_fields
+        Any additional fields to supply to the summarizer.
+    copy_files
+        Files to copy to the runtime directory.
+
+    Returns
+    -------
+    RunSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][]
+    """
+    preset = preset or "default"
+
+    # This will be useful if we add presets...
+    input_data = calc_swaps.get("input_data", {})
+    input_data = construct_namelist(input_data)
+    input_data = {k: dict(v) for k, v in input_data.items()}
+    input_data['control']['pseudo_dir'] = ESPRESSO_PP_PATH
+
+    if preset:
+        config = load_yaml_calc(ESPRESSO_PRESET_PATH / f"{preset}")
+        preset_pp = parse_pp_and_cutoff
```

Comment:
If we decide that a new calculator is indeed the best approach, I would recommend adding `preset` as a kwarg of the calculator and handling the parameter updating business there. That's what I did for the (slightly over-engineered) custom `Vasp` calculator [here](https://github.com/Quantum-Accelerators/quacc/blob/dca0cdec3be183877504d7849e073c81733d7ba1/src/quacc/calculators/vasp/vasp.py#L233-L243).

Reply:
I was hoping that a new calculator would not be needed for pw.x, but I understand that it is annoying to have all that in _base_job... I will come back to it after a first successful run

Reply:
I understand more now! This isn't a problem. If you end up making multiple "base" job functions, you can always abstract that part away into a separate function.